java/Nikiforov-Oleksandr/src/com/zoolatech/lecture7/tasks/_2/AtomicCalculator.java

```diff
@@ -2,29 +2,25 @@
 
 import java.util.concurrent.atomic.AtomicInteger;
 
-public class AtomicCalculator extends Calculator {
-    private final AtomicInteger atomicValue;
 
+public class AtomicCalculator extends Calculator {
     public AtomicCalculator(int value) {
         super(value);
-        this.atomicValue = new AtomicInteger(value);
     }
 
     @Override
     public void add(int valueToAdd) {
+        AtomicInteger atomicValue = new AtomicInteger(value);
         atomicValue.addAndGet(valueToAdd);
+        value = atomicValue.get();
```

Comment:
this doesn't prevent race conditions: two might enter this method in the same moment of time, and perform this operation concurrently

suggestion: remove this class from hierarchy and simply store the value in the AtomicInteger variable